yb-voyager/cmd/assessMigrationCommand.go

```diff
@@ -1816,7 +1816,65 @@ func generateAssessmentReportJson(reportDir string) error {
+		- Allows expanding to show detailed information, including category description, object type/name, SQL statement, supported versions, description, documentation link, and additional details.
+	- Handles cases where no issues are found, displaying an appropriate message.
+	- Utilizes helper functions such as `filterOutPerformanceOptimizationIssues`, `getPerformanceOptimizationIssues`, `snakeCaseToTitleCase`, `camelCaseToTitleCase`, and `getSupportedVersionString` for data formatting and filtering.
+
+	Usage:
+	------
+	- Include this template in a parent template using `{{ template "issuesTable" . }}`.
+	- Expects the following data structure in the context:
+		- .Issues: List of issue objects with fields like Category, Name, Impact, ObjectType, ObjectName, SqlStatement, Description, DocsLink, Details, MinimumVersionsFixedIn, CategoryDescription.
+		- .onlyPerf: Boolean flag indicating whether to show performance optimizations or general issues.
+
+		Differences Between the Two Tables Rendered by issuesTable
+		----------------------------------------------------------
+
+		The `issuesTable` template is used twice in the report: once for general assessment issues and once for performance optimizations. The differences between the two tables are as follows:
+
+		1. Heading and Labels:
+		- The heading is "Assessment Issues" for general issues and "Performance Optimizations" for performance-related issues.
+		- The count label is "Total issues" for general issues and "Total optimizations" for performance optimizations.
+		- The keyword in the table header is "Issue" or "Optimization" accordingly.
+
+		2. Data Source:
+		- For general issues, the table uses `filterOutPerformanceOptimizationIssues .Issues` to exclude performance optimizations.
+		- For performance optimizations, the table uses `getPerformanceOptimizationIssues .Issues` to include only those.
+
+		3. Table Columns:
+		- The general issues table includes a "Category" column (with an expand/collapse arrow).
+		- The performance optimizations table omits the "Category" column and places the expand/collapse arrow in the "Optimization" column.
+
+		4. Button IDs:
+		- The "Expand All" and "Collapse All" buttons have different IDs for each table to allow independent control.
 
+		5. Details Display:
+		- The details rows for general issues may include a "Category Description" field, which is omitted for performance optimizations.
+
+		6. Empty State:
+		- If there are no general issues, a message "No issues were found in the assessment." is shown.
+		- If there are no performance optimizations, no message is shown (the table is simply omitted).
+
+		7. Sorting:
+		- Sorting by "Category" is only available in the general issues table.
+
+		These differences are controlled by the `.onlyPerf` flag passed to the template and are reflected in both the Go template logic and the rendered HTML structure.
+
```

Comment:
Thanks for this detailed writeup.

Couple of questions:
1. Is there any change in UI before and after this change.
2. Internally in the json payload/report, how is performance optimisations categorized - is it a category of assessment issue, or different there also.

Reply:
1. No, verified manually
2. It is a category in the assessment issue in the JSON report or any other place. Just the HTML view is different